### PR TITLE
Disable the `MaxListenersExceededWarning` in Node when creating event targets for internal use

### LINKED ISSUE #3624: LogSubscription warning `MaxListenersExceededWarning: Possible EventTarget memory leak detected.`
## Overview

Hi all, I've started playing with the 2.x version of the SDK and pretty happy with it overall. I've now wanted to add 3 logsListeners with a filter of my address for the different commitment levels and while it works fine for 2 logs subscriptions, once I add a 3rd node events gets unhappy with 

When I reduce the `maxSubscriptionsPerChannel` config in the `createSolanaRpcSubscriptions` to say 2, the warning goes away fyi. The issue varies and sometimes complains about `AbortSignal` and sometimes about `EventTarget`

## Steps to reproduce

```typescript
const abortController = new AbortController();
const solanaRpcSubscriptions = createSolanaRpcSubscriptions(...);

const sub1 = solanaRpcSubscriptions.logsNotifications(
        { mentions: ["someAddress"] },
        { commitment: "processed" }).subscribe({abortSignal: abortController.signal}).then(...);
const sub2 = solanaRpcSubscriptions.logsNotifications(
        { mentions: ["someAddress"] },
        { commitment: "confirmed" }).subscribe({abortSignal: abortController.signal}).then(...);
const sub3 = solanaRpcSubscriptions.logsNotifications(
        { mentions: ["someAddress"] },
        { commitment: "finalized" }).subscribe({abortSignal: abortController.signal}).then(...);

```

## Description of bug

I did not expect to see this warning and more so would like for a way to set the limits such that it does not warn

EventTarget warning:
```bash
(node:54008) MaxListenersExceededWarning: Possible EventTarget memory leak detected. 11 error listeners added to EventTarget. MaxListeners is 10. Use events.setMaxListeners() to increase limit
    at [kNewListener] (node:internal/event_target:566:17)
    at EventTarget.addEventListener (node:internal/event_target:679:23)
    at Object.on (/Users/chris/Projects/spot/node_modules/@solana/subscribable/src/data-publisher.ts:28:26)
    at Object.on (/Users/chris/Projects/spot/node_modules/@solana/rpc-subscriptions-spec/src/rpc-subscriptions-channel.ts:38:32)
    at Object.on (/Users/chris/Projects/spot/node_modules/@solana/rpc-subscriptions-spec/src/rpc-subscriptions-pubsub-plan.ts:207:36)
    at Object.createAsyncIterableFromDataPublisher (/Users/chris/Projects/spot/node_modules/@solana/subscribable/src/async-iterable.ts:87:19)
    at Object.subscribe (/Users/chris/Projects/spot/node_modules/@solana/rpc-subscriptions-spec/src/rpc-subscriptions.ts:72:20)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

AbortSignal warning:
```bash
(node:54231) MaxListenersExceededWarning: Possible EventTarget memory leak detected. 11 abort listeners added to [AbortSignal]. MaxListeners is 10. Use events.setMaxListeners() to increase limit
    at [kNewListener] (node:internal/event_target:566:17)
    at [kNewListener] (node:internal/abort_controller:240:24)
    at EventTarget.addEventListener (node:internal/event_target:679:23)
    at Object.createAsyncIterableFromDataPublisher (/Users/chris/Projects/spot/node_modules/@solana/subscribable/src/async-iterable.ts:81:17)
    at Object.subscribe (/Users/chris/Projects/spot/node_modules/@solana/rpc-subscriptions-spec/src/rpc-subscriptions.ts:72:20)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

```

### PR DESCRIPTION
**Pull request by steveluscher**
# Summary

Read all about why, here: https://solana.stackexchange.com/a/17971/75

Fixes #3624.

# Test Plan

```ts
const {createSolanaRpcSubscriptions} = require('./dist/index.node.cjs');

const abortController = new AbortController();
const solanaRpcSubscriptions = createSolanaRpcSubscriptions('ws://localhost:8900');

const sub1 = solanaRpcSubscriptions.logsNotifications(
        { mentions: ['1'.repeat(32)] },
        { commitment: "processed" }).subscribe({abortSignal: abortController.signal}).then(() => {});
const sub2 = solanaRpcSubscriptions.logsNotifications(
        { mentions: ['1'.repeat(32)] },
        { commitment: "confirmed" }).subscribe({abortSignal: abortController.signal}).then(() => {});
const sub3 = solanaRpcSubscriptions.logsNotifications(
        { mentions: ['1'.repeat(32)] },
        { commitment: "finalized" }).subscribe({abortSignal: abortController.signal}).then(() => {});
```

No more warning.

### CODE REVIEW

**Comment by steveluscher** [comment]
* **#3661** <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3661?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3661?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `master`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by steveluscher — COMMENTED** [review]
What do you think of this? It's pretty heavy handed, but probably correct given how we use event targets internally to implement publish/subscribe/demux.

File: packages/event-target-impl/src/index.node.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { setMaxListeners } from 'node:events';
+
+export const AbortController = class extends globalThis.AbortController {
+    constructor(...args: ConstructorParameters<typeof globalThis.AbortController>) {
+        super(...args);
+        setMaxListeners(Number.MAX_SAFE_INTEGER, this.signal);
+    }
+};
```
Comment (lorisleiva) [inline packages/event-target-impl/src/index.node.ts:8]: What's the reason behind setting this inside the `AbortController` constructor as well?

**Review by lorisleiva — APPROVED** [review]
Makes sense, I didn't know Node would start throwing warning at 10 listeners haha. Thanks for the fix!

File: packages/event-target-impl/src/index.node.ts (hunk 1 above)
Comment (steveluscher) [reply packages/event-target-impl/src/index.node.ts:8]: `signal` is an `EventTarget`, and apparently attaching `addEventListener('abort', ...)` to it more than 10 times also raises this warning.

**Comment by steveluscher** [comment]
### Merge activity

* **Dec 4, 9:49 AM PST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/3661).

**Pull request merged**